### DISCUSSION TITLE
Copilot CLI support for Fish

### DISCUSSION THREAD

**Discussion by nickserv** [community/community#50792] [Copilot Conversations]
### Select Topic Area

Bug

### Body

[Fish](https://fishshell.com/) is a high quality, easy to use shell for Mac and Linux. It includes features like autocomplete, highlighting, and improved command history that would make Copilot CLI easier to use. However, the installation command does not expose any commands, probably because some scripts need to be ported to Fish.

**Comment by Akash1134** [comment]
Hi @nickmccurdy , Thank you for your feedback. We're letting users try out Copilot for CLI as a prototype!

Please [sign up for the waitlist](https://github.com/login/oauth/authorize?client_id=Iv1.43d3cd82bfd1c7a5&state=ZDGI6sJuplsGLDxmjBJHT&redirect_uri=https%3A%2F%2Fgithubnext.com%2Fsignup%2Fauth_callback), and we'll notify you when you're admitted. Note that you will also need GitHub Copilot access to use it.

If you need more information, you may also check out [this article](https://www.npmjs.com/package/@githubnext/github-copilot-cli?activeTab=readme).

**Reply by nickserv** [reply]
I'm already in the preview, that's how I know it isn't working.

**Reply by Akash1134** [reply]
Thank you for getting back @nickmccurdy. We have shared your feedback with the concerned team.

**Comment by dreki** [comment]
@nickmccurdy I’ve just written these. Please enjoy (at your own risk!).

Put these in `~/.config/fish/functions`

# qq.fish

```
function gq --wraps github-copilot-cli --description "Alias github-copilot-cli git-assist"
    set TMPFILE $(mktemp)
    trap "rm -f $TMPFILE" EXIT
    if /opt/homebrew/bin/github-copilot-cli git-assist "$argv" --shellout $TMPFILE
        if test -e $TMPFILE
            set FIXED_CMD $(cat $TMPFILE)
            eval $FIXED_CMD
        else
            echo "Apologies! Extracting command failed"
        end
    else
        return 1
    end
end
```

# gq.fish

```
function gq --wraps github-copilot-cli --description "Alias github-copilot-cli git-assist"
    set TMPFILE $(mktemp)
    trap "rm -f $TMPFILE" EXIT
    if /opt/homebrew/bin/github-copilot-cli git-assist "$argv" --shellout $TMPFILE
        if test -e $TMPFILE
            set FIXED_CMD $(cat $TMPFILE)
            eval $FIXED_CMD
        else
            echo "Apologies! Extracting command failed"
        end
    else
        return 1
    end
end
```

# ghq.fish

```
function ghq --wraps github-copilot-cli --description "Alias github-copilot-cli gh-assist"
    set TMPFILE $(mktemp)
    trap "rm -f $TMPFILE" EXIT
    if /opt/homebrew/bin/github-copilot-cli gh-assist "$argv" --shellout $TMPFILE
        if test -e $TMPFILE
            set FIXED_CMD $(cat $TMPFILE)
            eval $FIXED_CMD
        else
            echo "Apologies! Extracting command failed"
        end
    else
        return 1
    end
end
```

**Reply by dreki** [reply]
qq — `??`
gq — `git?`
ghq — `gh?`

**Reply by DiabolusGX** [reply]
Small correction in `qq.fish`
Looks like `qq` and `gq` are duplicate, sharing fix -

### qq.fish

```
function qq --wraps github-copilot-cli --description "Alias
github-copilot-cli what-the-shell"
    set TMPFILE $(mktemp)
    trap "rm -f $TMPFILE" EXIT
    if /opt/homebrew/bin/github-copilot-cli what-the-shell "$argv"
--shellout $TMPFILE
        if test -e $TMPFILE
            set FIXED_CMD $(cat $TMPFILE)
            eval $FIXED_CMD
        else
            echo "Apologies! Extracting command failed"
        end
    else
        return 1
    end
end
```

**Comment by z11i** [comment]
I made a plugin for Fish that you can install with

```
fisher install z11i/github-copilot-cli.fish
```

More details here: https://github.com/z11i/github-copilot-cli.fish
